packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1125,21 +1126,119 @@ class Positioned extends ParentDataWidget<StackRenderObjectWidgetBase> {
   }
 }
 
+abstract class GridRenderObjectWidgetBase extends MultiChildRenderObjectWidget {
+  GridRenderObjectWidgetBase({
+    List<Widget> children,
+    Key key
+  }) : super(key: key, children: children);
+
+  /// The delegate that controls the layout of the children.
+  GridDelegate get delegate;
+
+  RenderGrid createRenderObject() => new RenderGrid(delegate: delegate);
+
+  void updateRenderObject(RenderGrid renderObject, GridRenderObjectWidgetBase oldWidget) {
+    renderObject.delegate = delegate;
+  }
+}
+
 /// Uses the grid layout algorithm for its children.
 ///
 /// For details about the grid layout algorithm, see [RenderGrid].
-class Grid extends MultiChildRenderObjectWidget {
-  Grid(List<Widget> children, { Key key, this.maxChildExtent })
+class CustomGrid extends GridRenderObjectWidgetBase {
+  CustomGrid(List<Widget> children, { Key key, this.delegate })
     : super(key: key, children: children) {
-    assert(maxChildExtent != null);
+    assert(delegate != null);
+  }
+
+  /// The delegate that controls the layout of the children.
+  final GridDelegate delegate;
+}
+
+/// Uses a grid layout with a fixed column count.
+///
+/// For details about the grid layout algorithm, see [MaxTileWidthGridDelegate].
+class FixedColumnCountGrid extends GridRenderObjectWidgetBase {
+  FixedColumnCountGrid(List<Widget> children, {
+    Key key,
+    this.columnCount,
+    this.tileAspectRatio: 1.0,
+    this.padding: EdgeDims.zero
+  }) : super(key: key, children: children) {
+    assert(columnCount != null);
   }
 
-  final double maxChildExtent;
+  /// The number of columns in the grid.
+  final int columnCount;
 
-  RenderGrid createRenderObject() => new RenderGrid(maxChildExtent: maxChildExtent);
+  /// The ratio of the width to the height of each tile in the grid.
+  final double tileAspectRatio;
+
+  /// The amount of padding to apply to each child.
+  final EdgeDims padding;
+
+  FixedColumnCountGridDelegate get delegate {
+    return new FixedColumnCountGridDelegate(
```

Comment:
Does the delegate getter get called only once per FixedColumnCountGrid instance creation?